test/enhance/test_jpeg.py

```diff
@@ -0,0 +1,990 @@
+                            0.060,
+                        ],
+                        [
+                            0.063,
+                            0.009,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            0.009,
+                            0.063,
+                        ],
+                        [
+                            0.002,
+                            0.015,
+                            0.008,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            -0.000,
+                            0.008,
+                            0.015,
+                            0.002,
+                        ],
+                        [
+                            -0.000,
+                            0.002,
+                            0.063,
+                            0.060,
+                            0.020,
+                            0.017,
+                            0.078,
+                            0.146,
+                            0.146,
+                            0.078,
+                            0.017,
+                            0.020,
+                            0.060,
+                            0.063,
+                            0.002,
+                            -0.000,
+                        ],
+                    ],
+                ]
+            ],
+            device=device,
+            dtype=dtype,
+        )
+        # We use a slightly higher tolerance since our implementation varies from the reference implementation
+        assert_close(img_jpeg[..., :4, :4], img_jpeg_ref[..., :4, :4], rtol=0.01, atol=0.01)
+
+    def test_module(self, device, dtype) -> None:
+        B, H, W = 4, 16, 16
+        img = torch.rand(B, 3, H, W, device=device, dtype=dtype)
+        jpeg_quality = torch.randint(low=0, high=100, size=(B,), device=device, dtype=dtype)
+        qt_y = torch.randint(low=1, high=255, size=(B, 8, 8), device=device, dtype=dtype)
+        qt_c = torch.randint(low=1, high=255, size=(B, 8, 8), device=device, dtype=dtype)
+        diff_jpeg_module = kornia.enhance.DiffJPEG()
+        img_jpeg = diff_jpeg_module(img, jpeg_quality, qt_y, qt_c)
+        assert img_jpeg is not None
+        assert img_jpeg.shape == img.shape
+
+    def test_gradcheck(self, device) -> None:
```

Comment:
can you make somehow a test that performs a backward and check the gradients ? it would be funny to have a differentiable method without testing that part

Reply:
@edgarriba fully agree, however, PyTorch's grad check function is not really applicable here. I could check the gradient based on the reference implementation. What do you think about this?

Reply:
I added a test to check that the gradients are close to the reference implementation.